eclipse/src/saros/ui/actions/RequestSessionInviteAction.java

```diff
@@ -0,0 +1,86 @@
+package saros.ui.actions;
+
+import java.util.List;
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.ui.ISelectionListener;
+import org.eclipse.ui.IWorkbenchPart;
+import org.picocontainer.annotations.Inject;
+import saros.SarosConstants;
+import saros.SarosPluginContext;
+import saros.communication.extensions.JoinSessionRequestExtension;
+import saros.net.ITransmitter;
+import saros.net.xmpp.JID;
+import saros.net.xmpp.discovery.DiscoveryManager;
+import saros.session.ISarosSession;
+import saros.session.ISarosSessionManager;
+import saros.ui.Messages;
+import saros.ui.util.selection.SelectionUtils;
+import saros.ui.util.selection.retriever.SelectionRetrieverFactory;
+
+/**
+ * Action for requesting an invitation to a session from a contact.
+ *
+ * <p>This currently relies on the fact, that only Saros/S has a working JoinSessionRequestHandler.
+ * To make this feature generic in the future we need to add another XMPP namespace
+ */
+public class RequestSessionInviteAction extends Action implements Disposable {
+
+  @Inject private ISarosSessionManager sessionManager;
+  @Inject private DiscoveryManager discoveryManager;
+  @Inject private ITransmitter transmitter;
+
+  private ISelectionListener selectionListener =
+      new ISelectionListener() {
+        @Override
+        public void selectionChanged(IWorkbenchPart part, ISelection selection) {
+          updateActionState();
+        }
+      };
+
+  public static final String ACTION_ID = RequestSessionInviteAction.class.getName();
+
+  public RequestSessionInviteAction() {
+    super(Messages.RequestSessionInviteAction_title);
+    setId(ACTION_ID);
+    SarosPluginContext.initComponent(this);
+    SelectionUtils.getSelectionService().addSelectionListener(selectionListener);
+    updateActionState();
+  }
+
+  @Override
+  public void run() {
+    ISarosSession session = sessionManager.getSession();
+    JID jid = getSelectedJID();
+    if (session != null || jid == null) {
+      return;
+    }
+
+    // TODO: Currently only Saros/S is known to have a working JoinSessionRequestHandler,
+    //       remove this once the situation changes / change this to it's own feature.
+    if (!discoveryManager.isFeatureSupported(jid, SarosConstants.NAMESPACE_SERVER)) {
```

Comment:
Is there a reason to check twice whether the selected user is a Saros server (here and in line 445 of `SarosView`)? Or am I misunderstanding the checks?

This check is also missing the null-check for the result of ´discoveryManager.isFeatureSupported(...)`.

Reply:
removed